File: Java/TemplateUI/app/src/main/java/com/example/dl_sdk_sample_app/Func0Fragment.java
package com.example.dl_sdk_sample_app;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import com.example.dl_sdk_sample_app.databinding.FragmentFunc0Binding;
import com.datalogic.decode.BarcodeManager;
import com.datalogic.decode.ReadListener;

public class Func0Fragment extends Fragment {

    private FragmentFunc0Binding binding;
    private BarcodeManager barcodeManager;
    private ReadListener readListener;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = FragmentFunc0Binding.inflate(inflater, container, false);

        binding.scanButton.setOnClickListener(v -> initScanner());

        return binding.getRoot();
    }

    private void initScanner() {
        barcodeManager = new BarcodeManager();
        readListener = decodeResult -> getActivity().runOnUiThread(() -> {
            binding.resultTextView.setText(decodeResult.getText());
            Toast.makeText(getContext(), "Barcode Scanned: " + decodeResult.getText(), Toast.LENGTH_SHORT).show();
        });

        barcodeManager.addReadListener(readListener);
        Toast.makeText(getContext(), "Scanning Started...", Toast.LENGTH_SHORT).show();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        if (barcodeManager != null && readListener != null) {
            barcodeManager.removeReadListener(readListener);
        }
        binding = null;
    }
}

File: Kotlin/DL_SDK_Sample_App/app/src/main/java/com/example/dl_sdk_sample_app/Func0Fragment.kt
package com.example.dl_sdk_sample_app

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.fragment.app.Fragment
import com.example.dl_sdk_sample_app.databinding.FragmentFunc0Binding
import com.datalogic.decode.BarcodeManager
import com.datalogic.decode.ReadListener

class Func0Fragment : Fragment() {

    private var _binding: FragmentFunc0Binding? = null
    private val binding get() = _binding!!

    private var barcodeManager: BarcodeManager? = null
    private var readListener: ReadListener? = null

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentFunc0Binding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        // Set up the scan button click listener
        binding.scanButton.setOnClickListener {
            initScanner()
        }
    }

    private fun initScanner() {
        // Initialize BarcodeManager
        barcodeManager = BarcodeManager()

        // Define the ReadListener
        readListener = ReadListener { decodeResult ->
            activity?.runOnUiThread {
                binding.resultTextView.text = decodeResult.text
                Toast.makeText(requireContext(), "Barcode Scanned: ${decodeResult.text}", Toast.LENGTH_SHORT).show()
            }
        }

        // Add the ReadListener to BarcodeManager
        barcodeManager?.addReadListener(readListener)

        // Optionally, show a message indicating scanning has started
        Toast.makeText(requireContext(), "Scanning Started...", Toast.LENGTH_SHORT).show()
    }

    override fun onDestroyView() {
        super.onDestroyView()
        // Clean up resources
        barcodeManager?.removeReadListener(readListener)
        barcodeManager = null
        readListener = null
        _binding = null
    }
}
